### PR TITLE
Teach wmlscope to ignore frame durations. Fixes #2045

### LINKED ISSUE #2045: wmlscope doesn't handle multiple files in single line if they have frame duration specified
When you run `wmlscope --unresolved wesnoth/data/core` you get multiple false positives such as:
```"units/elves/Shaman.cfg", line 73: Unresolved reference -> 225,units/elves-wood/shaman-attack2.png
"units/elves/Shaman.cfg", line 73: Unresolved reference -> 225,units/elves-wood/shaman.png
"units/elves/Shaman.cfg", line 74: Unresolved reference -> 75,halo/elven/nature-halo7.png
"units/elves/Shaman.cfg", line 74: Unresolved reference -> 75,halo/elven/nature-halo8.png
```
The code in place looks like:
```wml
[frame]
    image="units/elves-wood/shaman.png:225,units/elves-wood/shaman-attack2.png:225,units/elves-wood/shaman.png:100"
    halo=halo/elven/nature-halo[1~6].png:75,halo/elven/nature-halo[7~8].png:50
    halo_x,halo_y=0,-12
[/frame]
```
So it seems that wmlscope doesn't recognize frame durations.

### PR DESCRIPTION
**Pull request by AI0867**

### CODE REVIEW

File: data/tools/wesnoth/wmltools3.py [hunk 1]
```diff
@@ -25,6 +25,28 @@ def __init__(self, sub, start, end):
     def __repr__(self):
         return "<Class Substitution, sub={}, start={:d}, end={:d}>".format(self.sub, self.start, self.end)
 
+def split_filenames(match):
+    filename_group = match.group(1)
+    bracket = filename_group.find('[')
+    comma = filename_group.find(',')
+    if not comma:
```
Comment (Elvish-Hunter) [inline data/tools/wesnoth/wmltools3.py]: I'd remove the `comma` variable and use `if ',' not in filename_group:`

File: data/tools/wesnoth/wmltools3.py [hunk 2]
```diff
@@ -25,6 +25,28 @@ def __init__(self, sub, start, end):
...
+        yield filename_group
+    elif not bracket:
```
Comment (Elvish-Hunter) [inline data/tools/wesnoth/wmltools3.py]: I'd remove the `bracket` variable and use `if '[' not in filename_group:`

File: data/tools/wesnoth/wmltools3.py [hunk 3]
```diff
@@ -477,7 +499,7 @@ def __str__(self):
 
 class CrossRef:
     macro_reference = re.compile(r"\{([A-Z_][A-Za-z0-9_:]*)(?!\.)\b")
-    file_reference = re.compile(r"[A-Za-z0-9{}.][A-Za-z0-9_/+{}.@\-\[\],~\*]*(" + "|".join(resource_extensions) + ")(?=(~.*)?)")
+    file_reference = re.compile(r"([A-Za-z0-9{}.][A-Za-z0-9_/+{}.@\-\[\],~\*]*(" + "|".join(resource_extensions) + "))((~[A-Z]+\(.*\))*)(:([0-9]+|\[[0-9,*~]*\]))?")
```
Comment (Elvish-Hunter) [inline data/tools/wesnoth/wmltools3.py]: While testing the former regex, I noticed that it had issues with multiple filenames on the same line even if durations weren't specified. For this reason, I'd try making the second character set non greedy (`*?`) and adding a match to a literal dot after it (`\.`).

Comment (AI0867) [reply data/tools/wesnoth/wmltools3.py]: I agree that the old regex had issues even without durations. I don't quite see what change you are suggesting or what difference it should make though.

Comment (AI0867) [reply data/tools/wesnoth/wmltools3.py]: Ah, you mean adding a mandatory period in front of the extension? I suppose it would help.

**Pull request merged**